cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -0,0 +1,158 @@
+            key = parsed_message[1]
+            logs.append(aws.get_object_content_from_s3(bucket, key))
+
+        # Parse logs for on/off events
+        for log in logs:
+            if log is None or log == '':
+                continue
+            instances, instance_events = self._parse_log_for_ec2_events(log)
+            result = 'Found instances: {i} and events {e}'.format(
+                i=instances, e=instance_events)
+            self.stdout.write(_(result))
+
+        # TODO: Save the messages to the DB
+        # TODO: If we haven't seen this instance before we'll
+        #       need to run describe_instances on it.
+        #       Start and stop requests don't have all the
+        #       info we need.
+
+        # aws.delete_message_from_queue(queue_url, messages)
+
+    def _parse_sqs_message(self, message):
+        """
+        Parse SQS message for path to log in S3.
+
+        Args:
+            message (boto3.SQS.Message): The Message object.
+
+        Returns:
+            list(tuple): List of (bucket, key) log file locations in S3.
+
+        """
+        parsed_records = []
+        message_body = json.loads(message.body)
+        for record in message_body.get('Records', []):
+            bucket_name = record['s3']['bucket']['name']
+            object_key = record['s3']['object']['key']
+            parsed_records.append((bucket_name, object_key))
+        return parsed_records
+
+    def _parse_log_for_ec2_events(self, log):
+        """
+        Parse S3 log for EC2 on/off events.
+
+        Args:
+            log (string): The string contents of the log file.
+
+        Returns:
+            list(tuple): List of instance data seen in log.
+            list(tuple): List of instance_event data seen in log.
+
+        """
+        ec2_event_map = {
+            'RunInstances': 'power_on',
+            'StartInstances': 'power_on',
+            'StopInstances': 'power_off',
+            'TerminateInstances': 'power_off'
+        }
+        instances = []
+        instance_events = []
+        log = json.loads(log)
+
+        for record in log.get('Records', []):
+            if not self._is_on_off_event(record, ec2_event_map.keys()):
+                continue
+
+            account_id = record.get('userIdentity', {}).get('accountId')
+            event_type = ec2_event_map[record.get('eventName')]
+            region = record.get('awsRegion')
+            occured_at = record.get('eventTime')
+            ec2_info = record.get('responseElements', {})\
+                .get('instancesSet', {})\
+                .get('items', [])
+
+            for item in ec2_info:
+                instance_id = item.get('instanceId')
+                instance_type = item.get('instanceType')
+                ec2_ami_id = item.get('imageId')
+                subnet = item.get('subnetId')
+
+                instances.append((account_id, instance_id, region))
+
+                # Subnet, ec2_ami_id, and instance_type will be None
```

Comment:
Why? its kinda important for us to know the those things when the instance starts/stops. We need to get that info somehow I think.

Reply:
As to why, I can't speak for AWS; they just don't include the info for those calls. But, I agree. There's a TODO above to run an describe_instances call to get that info. I was in a bit of a Catch 22 and decided this is a task for the next story where we actually save info the DB. 

I can re-post the comment here, but I commented directly on #28 with the two paths I think we can take there.